ext/ffi/jit.rs

```diff
@@ -0,0 +1,247 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use crate::tcc::Context;
+use crate::NativeType;
+use crate::Symbol;
+use deno_core::error::AnyError;
+use deno_core::v8;
+use std::ffi::c_void;
+use std::ffi::CString;
+
+macro_rules! cstr {
+  ($st:expr) => {
+    &CString::new($st).unwrap()
+  };
+}
+
+fn native_to_c(ty: &NativeType) -> &'static str {
+  match ty {
+    NativeType::Void => "void",
+    NativeType::U8 => "unsigned char",
+    NativeType::U16 => "unsigned short",
+    NativeType::U32 => "unsigned int",
+    NativeType::U64 | NativeType::USize => "unsigned long",
```

Comment:
I'm definitely not an expert on C and had to google but internet seems to suggest that `int` and `long int` are both 4 bytes, whereas `long long` is 8 bytes. I have no idea if that's correct but I'll take the risk of asking an idiotic question: Are these correct?

Also, should these perhaps allow for 32bit systems?

Reply:
Rust's `c_long` definition claims that on Windows this'll be 4 bytes but on Linux (often) 8 bytes.